src/test/java/org/rulez/demokracia/pdengine/VoteShowTest.java

```diff
@@ -11,110 +14,154 @@
+	private static final String CAN_ADDIN = "canAddin";
+	private static final String VOTE_PARAMETERS = "parameters";
+	private static final String NAME = "name";
+	private static final String USER = "user";
+	private static final String CHOICE_NAME = "choiceName";
 	private Vote vote;
 	private String voteId;
 	private String adminKey;
 
+	@Override
 	@Before
 	public void setUp() {
 		super.setUp();
 		vote = voteManager.getVote(adminInfo.voteId);
-		vote.addChoice("choiceName", "user");
-		voteManager.addChoice(new VoteAdminInfo(adminInfo.voteId, adminInfo.adminKey), "choiceName", "user");
+		vote.addChoice(CHOICE_NAME, USER);
+		voteManager.addChoice(new VoteAdminInfo(adminInfo.voteId, adminInfo.adminKey), CHOICE_NAME, USER);
 		voteId = adminInfo.voteId;
 		adminKey = adminInfo.adminKey;
 	}
-	
+
 	@Test
 	public void the_name_attribute_contains_the_name_of_the_vote() {
-		JSONObject result = voteManager.showVote(new VoteAdminInfo(voteId, adminKey));
-		
-		assertEquals(result.get("name"), vote.name);
+		JsonObject result = voteManager.showVote(new VoteAdminInfo(voteId, adminKey));
+
+		assertEquals(result.get(NAME).getAsString(), vote.name);
 	}
-	
+
 	@Test
 	public void the_canAddIn_attribute_contains_whether_the_voters_can_add_choices_to_the_vote() {
-		JSONObject result = voteManager.showVote(new VoteAdminInfo(voteId, adminKey));
-		
-		assertEquals(result.get("canAddIn"), vote.parameters.canAddin);
+		JsonObject result = voteManager.showVote(new VoteAdminInfo(voteId, adminKey));
+
+		assertEquals(result.get(VOTE_PARAMETERS).getAsJsonObject().get(CAN_ADDIN).getAsBoolean(),
+				vote.parameters.canAddin);
 	}
 
 	@Test
 	public void the_creationTime_attribute_contains_the_creation_time_of_the_vote() {
-		JSONObject result = voteManager.showVote(new VoteAdminInfo(voteId, adminKey));
-		
-		assertEquals(result.get("creationTime"), vote.creationTime);
+		JsonObject result = voteManager.showVote(new VoteAdminInfo(voteId, adminKey));
+
+		assertEquals(result.get(CREATION_TIME).getAsLong(), vote.creationTime);
 	}
-	
+
 	@Test
 	public void the_choices_attribute_contains_the_choices_of_the_vote() {
-		JSONObject result = voteManager.showVote(new VoteAdminInfo(voteId, adminKey));
-		JSONArray jsonArray = vote.createChoicesJson(vote.choices);
-		
-		assertEquals(result.getJSONArray("choices").toString(), jsonArray.toString());
+		JsonObject result = voteManager.showVote(new VoteAdminInfo(voteId, adminKey));
+
+		assertEquals(vote.choices.size(), result.get(CHOICES).getAsJsonObject().entrySet().size());
 	}
 	@Test
 	public void the_initiator_of_the_choice_is_in_the_json() {
-		JSONArray jsonArray = vote.createChoicesJson(vote.choices);
-		JSONObject firstChoice = (JSONObject) jsonArray.get(0);
-		String initiatorInJson = (String) firstChoice.get("initiator");
-		assertEquals("user", initiatorInJson);
+		JsonObject choicesJson = vote.toJson().get(CHOICES).getAsJsonObject();
+		assertFalse(choicesJson.entrySet().isEmpty());
```

Comment:
Factor out this complex set of asserts to an assert function in the parent class.